tembo-pgmq-python/tests/test_integration.py

```diff
@@ -193,6 +193,55 @@ def test_validate_queue_name(self):
             self.queue.validate_queue_name(invalid_queue_name)
         self.assertIn("queue name is too long", str(context.exception))
 
+    def test_transaction_create_queue(self):
+        """Test creating a queue within a transaction."""
+        try:
+            self.queue.create_queue("test_queue_txn", perform_transaction=True)
+            raise Exception("Intentional failure")
+        except Exception:
+            pass
+        finally:
+            queues = self.queue.list_queues(perform_transaction=False)
+            self.assertNotIn("test_queue_txn", queues)
+
+    def test_transaction_send_and_read_message(self):
+        """Test sending and reading a message within a transaction."""
+        try:
+            self.queue.send(
+                self.test_queue, self.test_message, perform_transaction=True
+            )
+            raise Exception("Intentional failure")
+        except Exception:
+            pass
+        finally:
+            message = self.queue.read(self.test_queue, perform_transaction=False)
+            self.assertIsNone(message, "No message expected in queue")
+
+    def test_transaction_purge_queue(self):
+        """Test purging a queue within a transaction."""
+        self.queue.send(self.test_queue, self.test_message, perform_transaction=False)
+        try:
+            self.queue.purge(self.test_queue, perform_transaction=True)
+            raise Exception("Intentional failure")
+        except Exception:
+            pass
+        finally:
+            message = self.queue.read(self.test_queue, perform_transaction=False)
+            self.assertIsNotNone(message, "Message expected in queue")
+
+    def test_transaction_rollback(self):
+        """Test rollback of a transaction."""
+        try:
+            self.queue.send(
+                self.test_queue, self.test_message, perform_transaction=True
+            )
+            raise Exception("Intentional failure to trigger rollback")
```

Comment:
I think we need to raise something specific here. The way this is currently written, `self.queue.send()` can fail for any reason and the test will still pass.